packages/frontend-shared/src/gql-components/Auth.vue

```diff
@@ -119,99 +127,76 @@ mutation Auth_ResetAuthState {
-const clickedOnce = ref(false)
+onBeforeUnmount(() => {
+  // If a log in was initiated from this component instance, then the auth state
+  // may be polluted, due to the mutation still being fetched or due to
+  // errors returned during the login process. So a reset occurs when
+  // this instance unmounts to cover all scenarios where the LoginModal may be dismissed.
+  //
+  // We only perform the reset for the component that triggered a login
+  // to prevent state conflicts when LoginModals are presented within the launchpad
+  // and app simultaneously.
+  if (loginInitiated.value) {
+    reset.executeMutation({})
+  }
+})
 
 const emit = defineEmits<{
   (event: 'continue', value: boolean): void
 }>()
 
-const viewer = computed(() => props.gql.cloudViewer)
-const isBrowserOpened = computed(() => props.gql.authState.browserOpened)
-const isLoggingIn = computed(() => clickedOnce.value && !viewer.value)
+const cloudViewer = computed(() => {
+  return props.gql.cloudViewer
+})
+
+const browserOpened = computed(() => {
+  return props.gql.authState.browserOpened
+})
 
-const query = useQuery({
-  query: Auth_BrowserOpenedDocument,
-  requestPolicy: 'cache-and-network',
+// We determine that a login is pending if there is no current cloudViewer and
+// either a login has been initiated from this component, or the browser has been
+// successfully opened.
+//
+// It is possible for the browser to be open but not due to actions by this component,
+// particularly when LoginModals are presented in both the launchpad and app simultaneously.
+const loginMutationIsPending = computed(() => {
+  return !cloudViewer.value && (loginInitiated.value || browserOpened.value)
 })
 
-const handleAuth = async () => {
-  if (viewer.value) {
+const handleLoginOrContinue = async () => {
+  if (cloudViewer.value) {
     emit('continue', true)
 
     return
   }
 
-  clickedOnce.value = true
+  loginInitiated.value = true
 
-  const browserCheckInterval = setInterval(async () => {
-    await query.executeQuery({})
-    if (isBrowserOpened.value) {
-      clearInterval(browserCheckInterval)
-    }
-  }, 1500)
-
-  await login.executeMutation({})
+  login.executeMutation({})
 }
 
-const handleLogout = async () => {
-  await logout.executeMutation({})
+const handleLogout = () => {
+  logout.executeMutation({})
 }
 
-const buttonMessage = computed(() => {
-  if (!isBrowserOpened.value && isLoggingIn.value) {
-    return t('topNav.login.actionOpening')
-  }
-
-  if (!clickedOnce.value && !viewer.value) {
-    return t('topNav.login.actionLogin')
-  }
-
-  if (isLoggingIn.value) {
-    return t('topNav.login.actionWaiting')
-  }
-
-  if (viewer.value) {
-    return t('topNav.login.actionContinue')
-  }
-
-  // default
-  return t('topNav.login.actionLogin')
-})
-
-const buttonVariant = computed(() => {
-  if (clickedOnce.value && !viewer.value) {
-    return 'pending'
-  }
-
-  return 'primary'
-})
+const handleTryAgain = async () => {
+  await reset.executeMutation({})
```

Comment:
Minor note, we're now waiting for the the reset mutation to complete prior to kicking off the login. I'm not sure how these are enqueued, it might work out, but this makes our expectations explicit.